cfme/storage/volume_snapshot.py

```diff
@@ -0,0 +1,234 @@
+
+
+@attr.s
+class VolumeSnapshot(BaseEntity, WidgetasticTaggable):
+    """ Model of an Storage Volume Snapshots in cfme
+
+    Args:
+        name: name of the snapshot
+        provider: provider
+    """
+    name = attr.ib()
+    provider = attr.ib()
+
+    def refresh(self):
+        self.provider.refresh_provider_relationships()
+        self.browser.refresh()
+
+    @property
+    def exists(self):
+        """ check for snapshot exist on UI.
+
+        Returns:
+            :py:class:`bool`
+        """
+        view = navigate_to(self.parent, 'All')
+        return self.name in view.entities.all_entity_names
+
+    @property
+    def status(self):
+        """ status of cloud volume snapshot.
+
+        Returns:
+            :py:class:`str` Status of volume snapshot.
+        """
+        view = navigate_to(self.parent, 'All')
+        view.toolbar.view_selector.select("List View")
+
+        for item in view.entities.elements.read():
+            if self.name in item['Name']:
+                return str(item['Status'])
+
+    @property
+    def size(self):
+        """ size of cloud volume snapshot.
+
+        Returns:
+            :py:class:`int` size of volume snapshot in GB.
+        """
+        view = navigate_to(self, 'Details')
+        return int(view.entities.properties.get_text_of('Size').split()[0])
+
+    @property
+    def volume_name(self):
+        """ volume name of snapshot.
+
+        Returns:
+            :py:class:`str` respective volume name.
+        """
+        view = navigate_to(self, 'Details')
+        return view.entities.relationships.get_text_of('Cloud Volume')
+
+    @property
+    def tenant_name(self):
+        """ Tenant name of snapshot.
+
+        Returns:
+            :py:class:`str` respective tenant name for snapshot.
+        """
+        view = navigate_to(self, 'Details')
+        return view.entities.relationships.get_text_of('Cloud Tenants')
+
+    def delete(self):
+        """Delete snapshot """
+
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Delete Cloud Volume Snapshot')
+        view.flash.assert_success_message('Delete initiated for 1 Cloud Volume Snapshot.')
+
+        wait_for(
+            lambda: not self.exists,
+            message="Wait snapshot to disappear",
+            delay=20,
+            timeout=800,
+            fail_func=self.refresh
+        )
+
+
+@attr.s
+class VolumeSnapshotCollection(BaseCollection):
+    """Collection object for :py:class:`cfme.storage.volume_snapshots.VolumeSnapshot`"""
+
+    ENTITY = VolumeSnapshot
+
+    def all(self):
+        """returning all Snapshot objects for respective storage manager type"""
+        view = navigate_to(self, 'All')
+        view.toolbar.view_selector.select("List View")
+        snapshots = []
+
+        for item in view.entities.elements.read():
+            if self.filters.get('provider').name in item['Cloud Provider']:
```

Comment:
This needs to handle the AttributeException that occurs when there is no `provider` filter. 

Same behavior as: https://github.com/ManageIQ/integration_tests/issues/6123

Reply:
@mshriver can you please review #6132, so I can implement here also .has_key('provider') as well in issue you assign by you.